lib/rouge/lexers/mason.rb

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Mason < TemplateLexer
+      title 'Mason'
+      desc 'HTML::Mason'
+      tag 'mason'
+      filenames '*.mi', '*.mc', '*.mas', '*.m', '*.mhtml', '*.mcomp', 'autohandler', 'dhandler'
+      mimetypes 'text/x-mason', 'application/x-mason'
+
+      def initialize(*)
+        super
+        @perl = Perl.new
+      end
+  
+      def self.detect?(text)
+        return false if text.doctype?(/[ht|x]ml/)
+        return true if text.doctype?
+      end
+
+      # Note: If you add a tag in the lines below, you also need to modify "disambiguate '*.m'" in file disambiguation.rb
+      textblocks = %w(text doc)
+      perlblocks = %w(args flags attr init once shared perl cleanup filter)
+      components = %w(def method)
+  
+      state :root do
+        mixin :mason_tags
+      end
+
+      state :mason_tags do
+        rule %r/\s+/, Text::Whitespace
+
+        rule %r/<%(#{textblocks.join('|')})>/i, Comment::Preproc, :text_block

```

Comment:
Interpolated regexes are a source of memory drain. Hence prefer using them with the `o` flag where applicable:
```suggestion
        rule %r/<%(#{textblocks.join('|')})>/oi, Comment::Preproc, :text_block
```

Reply:
Nice to know! Applied.